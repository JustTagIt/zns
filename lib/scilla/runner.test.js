const runner = require('./runner')
const fs = require('fs')

it('should deploy without state or message', () => {
  expect(
    runner({
      init: [
        {
          vname: '_scilla_version',
          type: 'Uint32',
          value: '0',
        },
        {
          vname: '_this_address',
          type: 'ByStr20',
          value: '0xabfeccdc9012345678901234567890f777567890',
        },
        {
          vname: '_creation_block',
          type: 'BNum',
          value: '1',
        },
        {
          vname: 'owner',
          type: 'ByStr20',
          value: '0xabfeccdc9012345678901234567890f777567890',
        },
      ],
      blockchain: [{vname: 'BLOCKNUMBER', type: 'BNum', value: '100'}],
      gaslimit: 10000,
      libdirs: ['/Users/laptop3/ZilliqaNameService/scilla/src/stdlib'],
      code: fs.readFileSync(
        '/Users/laptop3/ZilliqaNameService/scilla/tests/contracts/helloWorld.scilla',
      ),
    }),
  ).toMatchSnapshot()
})

it('should run function with state and message', () => {
  expect(
    runner({
      init: [
        {
          vname: '_scilla_version',
          type: 'Uint32',
          value: '0',
        },
        {
          vname: '_this_address',
          type: 'ByStr20',
          value: '0xabfeccdc9012345678901234567890f777567890',
        },
        {
          vname: '_creation_block',
          type: 'BNum',
          value: '1',
        },
        {
          vname: 'owner',
          type: 'ByStr20',
          value: '0xabfeccdc9012345678901234567890f777567890',
        },
      ],
      blockchain: [{vname: 'BLOCKNUMBER', type: 'BNum', value: '100'}],
      state: [
        {
          vname: '_balance',
          type: 'Uint128',
          value: '0',
        },
      ],
      message: {
        _tag: 'setHello',
        _amount: '0',
        _sender: '0x1234567890123456789012345678901234567890',
        params: [
          {
            vname: 'msg',
            type: 'String',
            value: 'Hello World',
          },
        ],
      },
      gaslimit: 1000,
      libdirs: ['/Users/laptop3/ZilliqaNameService/scilla/src/stdlib'],
      code: fs.readFileSync(
        '/Users/laptop3/ZilliqaNameService/scilla/tests/contracts/helloWorld.scilla',
      ),
    }),
  ).toMatchSnapshot()
})
