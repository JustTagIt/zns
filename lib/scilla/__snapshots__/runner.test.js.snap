// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should deploy without state or message 1`] = `
Object {
  "_accepted": "false",
  "events": Array [],
  "gas_remaining": "8262",
  "message": null,
  "scilla_major_version": "0",
  "states": Array [
    Object {
      "type": "Uint128",
      "value": "0",
      "vname": "_balance",
    },
    Object {
      "type": "String",
      "value": "",
      "vname": "welcome_msg",
    },
  ],
}
`;

exports[`should run function with state and message 1`] = `
Object {
  "_accepted": "false",
  "events": Array [
    Object {
      "_eventname": "setHello()",
      "params": Array [
        Object {
          "type": "Int32",
          "value": "1",
          "vname": "code",
        },
      ],
    },
  ],
  "gas_remaining": "707",
  "message": null,
  "scilla_major_version": "0",
  "states": Array [
    Object {
      "type": "Uint128",
      "value": "0",
      "vname": "_balance",
    },
    Object {
      "type": "String",
      "value": "",
      "vname": "welcome_msg",
    },
  ],
}
`;
